src/FSharpVSPowerTools.Logic/GoToDefinitionFilter.fs

```diff
@@ -421,10 +432,21 @@ type GoToDefinitionFilter(textDocument: ITextDocument,
                 match symbolResult with
                 | Some (_, _, _, _, FSharpFindDeclResult.DeclFound _) 
                 | None ->
-                    // Run the operation on UI thread since continueCommandChain may access UI components.
-                    do! Async.SwitchToContext uiContext
-                    // Declaration location might exist so let Visual F# Tools handle it. 
-                    return continueCommandChain()
+                    // no FSharpSymbol found, here we look at #load directive
```

Comment:
Different computation expressions don't compose well.

You should use `asyncMaybe` monad. 
I think this should works:

```fsharp
let directive = 
    asyncMaybe {
         let! filepath, project, _ = getCurrentFilePathProjectAndDoc()
         return! vsLanguageService.GetLoadDirectiveFileNameAtCursor(filepath, view, project) |> liftAsync
      }
match directive with
| Some fileToOpen -> ...
| None -> ...
```

See another example at https://github.com/fsprojects/VisualFSharpPowerTools/blob/199bf001fab7d1bea0119e2661f34326d68db72a/src/FSharpVSPowerTools.Logic/QuickInfoMargin.fs#L90-L132